#!/usr/bin/env python
# vim: set syntax=python :

PACKAGE='platform_motion'
from dynamic_reconfigure.parameter_generator_catkin import ParameterGenerator, double_t

from math import pi

gen = ParameterGenerator()

gen.add( "port_steering_offset", double_t,
        0, "Port steering offset",
        0, -pi/10, pi/10)
gen.add( "starboard_steering_offset", double_t,
        0, "Starboard steering offset",
        0, -pi/10, pi/10)
gen.add( "stern_steering_offset", double_t,
        0, "Stern steering offset",
        0, -pi/10, pi/10)
gen.add( "carousel_offset", double_t,
        0, "carousle position offset",
        0, -pi, pi)
gen.add( "wheel_diameter",  double_t,
        0, "Vehicle wheel diameter",
        0.33, 0.2, 0.4)

exit(gen.generate(PACKAGE, "motion_node", "PlatformParameters"))
